File: admin-dashboard/src/models/Product.ts
export interface Product {
  productId: number;
  name: string;
  description: string;
  price: number;
  category: string;
  categoryId?: number;
  category_id?: number;
  image: string;
  isBestSeller: boolean;
  isTryFood: boolean;
  is_best_seller?: boolean;
  is_try_food?: boolean;
}

export interface ProductFormData {
  name: string;
  description: string;
  price: number;
  categoryId: number;
  image: string;
  is_best_seller: boolean;
  is_try_food: boolean;
} 

File: backend/kat/src/main/java/com/project/Kat/models/Product.java
package com.project.Kat.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "products")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Product extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "productId")
    private Long productId;

    @Column(name = "name", nullable = false, length = 350)
    private String name;

    private BigDecimal price;

    @Column(name = "image", length = 300)
    private String image;

    @Column(name = "description")
    private String description;

    @ManyToOne
    @JoinColumn(name = "category_id")
    @JsonBackReference
    private Category category;

    private Boolean isBestSeller = false;
    private Boolean isTryFood    = false;
}
